src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,67 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    private int randomeRadius = 0;
+    private int squareRandomeSide = 0;
+    private int rectangleRandomASide = 0;
+    private int rectangleRandomBSide = 0;
+    private int firstRandomeLeg = 0;
+    private int secondRandomeLeg = 0;
+    private int trapezoidRandomeASide = 0;
+    private int trapezoidRandomeBSide = 0;
+    private int trapezoidRandomeHeigth = 0;
```

Comment:
It's good that you don't pass expressions to constructors, instead using variables. 

But don't create them all on the class level. Instead move the logic of creating a random figure into the separate method and store those variables here.

[Reference](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#think-about-which-variables-should-be-local-in-the-method-and-which-should-be-class-level).